source/javascripts/components/_Trigger.js.erb

```diff
@@ -45,13 +45,19 @@
 				return "tag";
 			}
 		};
+		
+		Trigger.prototype.setTarget = function(target) {
+			delete this.triggerConfig.workflow;
+			delete this.triggerConfig.pipeline;
+			this.triggerConfig[target.type] = target.value;
+		};
 
-		Trigger.prototype.workflowID = function(workflowID) {
-			if (workflowID !== undefined) {
-				this.triggerConfig.workflow = workflowID;
+		Trigger.prototype.getTarget = function(target) {
+			var type = this.triggerConfig.workflow ? "workflow" : "pipeline"
+			return {
+				type: type,
+				value: this.triggerConfig[type]
```

Comment:
I'm not really a fan of that pattern, using the same method for two completely different purposes is rather confusing for me. Also setting the type itself is rather tricky as it needs a value at the same time. And we need to know where to set the value, so basically both of them need the other half to be viable.

Reply:
Does not seem that complex to me, they do use each other but I don't think that's a problem:

`.target`: if it's a setter, do `this.triggerConfig[this.targetType()] = target`; and for both setter & getter cases, return `this.triggerConfig[this.targetType()]`.

`.targetType`: if it's a setter, do `var target = this.triggerConfig.target()` then `delete this.triggerConfig[this.targetType()]`, finally `this.triggerConfig[targetType] = target`; and for both setter & getter cases, return either `workflow` or `pipeline`, whichever is defined as key in `this.triggerConfig`.

My problem is merging the two just as "target" is that now it is a non-primitive that is not documented anywhere (meaning `{type: string, value: string}`), instead of having 2 separately handled string "properties" (I mean getter-setters) for them. Also, whenever you want to set the type or the value, you need to define both of them, whereas if you use separate methods for them here, only one of them is needed.

As for the pattern, it tries to mimic how a regular property would behave: you can either get its value, or assign one to it. We use it throughout this project in case of all the other objects like `Step`, `Variable`, etc.

Reply:
I still think it's a horrible pattern, but I don't want to introduce inconsistencies with the other parts of the application. Maybe we could revisit the topic once we start migrating these parts to TS! :)